### PR TITLE
NIFI-4993: Add NiFi UserId to ReportLineageToAtlas

### PR DESCRIPTION
**Pull request by ijokarumawak**
## For reviewer(s)

Please be aware that there is another major issue with 'complete path' strategy, that is #2542. It should be easier to confirm this fix if that PR is cherry-picked as well when you review this PR (in case if that one is not merged yet to master branch when you review this one).

---

An existing NiFi user id who has sufficient privilege needs to be used
in order to query full lineage information for 'complete path' strategy
to work as expected.

Added new reporting task property so that user can specify such user id.

Also added exception handling logic to inform users that a proper user
id is required.

---

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/reporting/ReportLineageToAtlas.java [hunk 1]
```diff
@@ -584,6 +601,12 @@ public void onTrigger(ReportingContext context) {
             return;
         }
 
+        final String nifiUserId = context.getProperty(NIFI_USER_ID).evaluateAttributeExpressions().getValue();
+        if (isEmpty(nifiUserId)) {
+            getLogger().warn("NiFi user id is empty, make sure '" + NIFI_USER_ID.getDisplayName() + "' is configured correctly.");
+            return;
```
Comment (zenfenan) [inline nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/reporting/ReportLineageToAtlas.java]: Is this for backward compatibility? Reason why I'm asking is, if we are making `NIFI_USER_ID.required(true)` and setting appropriate validator, is this check required?

Comment (ijokarumawak) [reply nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/reporting/ReportLineageToAtlas.java]: @zenfenan Yes, it is. If NIFI_USER_ID is defined with NiFi Expression Language, evaluation results can be empty if EL refers non-existing properties ... etc.

**Comment by ijokarumawak** [comment]
I am now trying a different approach. Instead of letting user specify the user id, adding provenance query method that does not require any user for internal NiFi use. I will update this PR, when another one is ready.

**Comment by ijokarumawak** [comment]
This PR has been superseded by #2577.

**Pull request closed without merging**